### PR TITLE
change subtype to text

### PR DESCRIPTION
**Pull request by snopoke**
http://manage.dimagi.com/default.asp?158917

### CODE REVIEW

**Comment by snopoke** [comment]
@czue not sure if we want to merge this yet. The table will be locked while this migration happens which is likely to take a while since its going to re-write the entire table.

Optionally we can try and push this out until we can upgrade PostgreSQL.

**Comment by czue** [comment]
build fail looks real
```
======================================================================
ERROR: test_subtype_no_truncate (casexml.apps.stock.tests.test_stock_transaction.StockTransactionTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "corehq/ex-submodules/casexml/apps/stock/tests/test_stock_transaction.py", line 17, in test_subtype_no_truncate
    self._test_subtype(sub_type, sub_type)
  File "corehq/ex-submodules/casexml/apps/stock/tests/test_stock_transaction.py", line 48, in _test_subtype
    txn.save()
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/db/models/base.py", line 546, in save
    force_update=force_update, update_fields=update_fields)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/db/models/base.py", line 650, in save_base
    result = manager._insert([self], fields=fields, return_id=update_pk, using=using, raw=raw)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/db/models/manager.py", line 215, in _insert
    return insert_query(self.model, objs, fields, **kwargs)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/db/models/query.py", line 1675, in insert_query
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 943, in execute_sql
    cursor.execute(sql, params)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/db/backends/postgresql_psycopg2/base.py", line 58, in execute
    six.reraise(utils.DatabaseError, utils.DatabaseError(*tuple(e.args)), sys.exc_info()[2])
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/db/backends/postgresql_psycopg2/base.py", line 54, in execute
    return self.cursor.execute(query, args)
DatabaseError: value too long for type character varying(20)
----------------------------------------------------------------------
```

File: corehq/ex-submodules/casexml/apps/stock/models.py [hunk 1]
```diff
@@ -54,7 +65,7 @@ class StockTransaction(models.Model):
     # currently supported/expected: 'stockonhand', 'receipts', 'consumption'
     type = models.CharField(max_length=20)
     # e.g. 'loss', 'transfer', 'inferred'
-    subtype = TruncatingCharField(max_length=20, null=True, blank=True)
+    subtype = TruncatingTextField(max_length=40, null=True, blank=True)
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/stock/models.py]: if it's text is it still worth truncating?

Comment (czue) [inline corehq/ex-submodules/casexml/apps/stock/models.py]: (i'm genuinely conflicted on my feelings on this - not looking for a particular answer)

Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/stock/models.py]: No need to truncate from a DB point of view, only necessary if we want to protect against someone dumping a huge string in there but that's also likely taken care of in other places.

**Comment by czue** [comment]
fyi, build still failing

**Comment by snopoke** [comment]
Weird, can't get it to fail locally. Will have to spend more time figuring out what's going on.

**Comment by snopoke** [comment]
Note to self. Plan to do PG upgrade to make this migration easy. Will also then make this a normal Text field and add a check constraint in the migration.

**Comment by snopoke** [comment]
closing till pg upgrade done

**Pull request closed without merging**